src/Storage/FlysystemStorage.php

```diff
@@ -37,7 +37,7 @@ public function __construct(PropertyMappingFactory $factory, $registry)
         $this->registry = $registry;
     }
 
-    protected function doUpload(PropertyMapping $mapping, UploadedFile $file, ?string $dir, string $name): void
+    protected function doUpload(PropertyMapping $mapping, File $file, ?string $dir, string $name): void
```

Comment:
Missing check for `UploadedFile|ReplacingFile`

Reply:
This is a protected function and the check for the types is in the public method `upload()`.

Besides that, the methods all do work fine with a passed in `File` object. I do not see how restricting the type to a specialized one helps with anything? Quite the contrary, I think this is so deep in the abstraction stack, that it should not be relevant where the file comes from.

Actually, I think the whole `AbstractStorage` class should not depend on more than `File`. Not possible for `upload()` though, because it uses `getClientOriginalName()`, but all other methods work fine with an object of type `File` and I think this should be the way to go.

The check, if something is to be done, is already in the `UploadHandler` and I think duplicating this check everywhere across the source code just makes it harder to allow new file types, like the `ReplacingFile` type now.